site/src/content/docs/contribute/principles.mdx

```diff
@@ -0,0 +1,49 @@
+---
+title: Zarf Principles
+---
+
+Zarf is a tool for continuous software delivery to disconnected environments. To ensure that Zarf remains true to its mission and continues to provide value for its users, the following design principles serve as a framework for evaluating new features and capabilities.
+
+## 1. Airgap-First Functionality
+Zarf is designed for airgapped environments, meaning that all functionality should operate without requiring external dependencies. This includes:
+- Ensuring that all packaged software and dependencies are self-contained.
+- Avoiding reliance on external repositories or registries during runtime.
+- Providing a seamless user experience for both online and offline operations.
+
+## 2. Minimal and Secure Binary
+The Zarf binary must remain lightweight and secure while offering core functionality. Considerations include:
+- Baking in only essential tools necessary for package management, deployment, and validation.
+- Avoiding unnecessary bloat by carefully selecting embedded utilities.
+- Ensuring security best practices, including dependency verification and signed releases.
+
+## 3. Clear Scope Within Kubernetes Environments
+Zarf operates in Kubernetes clusters but does not attempt to replace existing Kubernetes tooling. Instead, Zarf should:
+- Focus on enabling package management and deployment in airgapped clusters.
+- Complement, rather than compete with, Kubernetes-native solutions (e.g., Helm, Kustomize, and Operator patterns).
+- Facilitate cluster initialization and ongoing software management without enforcing opinionated infrastructure choices.
+
+## 4. Declarative and Reproducible Workflows
+To maintain consistency in deployments, Zarf should emphasize:
+- Declarative package definitions that allow for repeatable deployments.
+- Versioned and auditable package formats that ensure integrity across different environments.
+- Support for GitOps workflows where applicable.
+
+## 5. Operational Simplicity
+Zarf should remain simple to use and operationalize, with considerations such as:
+- Clear and concise configuration files that reduce complexity.
+- A user-friendly CLI that abstracts away unnecessary details while allowing expert-level control when needed.
+- Automation and scripting capabilities that streamline usage in constrained environments.
+
+## 6. Compatibility and Extensibility
+Zarf should be flexible enough to work in various environments while maintaining its core mission. This means:
+- Supporting multiple architectures and operating systems where feasible.
+- Allowing extensibility through custom package configurations and integrations.
```

Comment:
Worth calling out its use as a library explicitly here? (since "integrations" could be many things and some of them we may not prefer (i.e. a python script that generates Zarf packages and drives Zarf's CLI would be an "integration" but IMO that would be treated differently from a Go CLI importing Zarf as a library)

Reply:
I give mention to the SDK now (Added) - there is likely room for further specification but we still need to establish that scope in other ways as well.

Reply:
yeah - I think the level you added is likely good for a principles doc and then other docs / information can help shape the rest.